apis/oam/v1alpha2/core_workload_types.go

```diff
@@ -0,0 +1,398 @@
+	Resources *ContainerResources `json:"resources,omitempty"`
+
+	// Command to be run by this container.
+	// +optional
+	Command []string `json:"command,omitempty"`
+
+	// Arguments to be passed to the command run by this container.
+	// +optional
+	Arguments []string `json:"args,omitempty"`
+
+	// Environment variables that should be set within this container.
+	// +optional
+	Environment []ContainerEnvVar `json:"env,omitempty"`
+
+	// ConfigFiles that should be written within this container.
+	// +optional
+	ConfigFiles []ContainerConfigFile `json:"config,omitempty"`
+
+	// Ports exposed by this container.
+	// +optional
+	Ports []ContainerPort `json:"ports,omitempty"`
+
+	// A LivenessProbe assesses whether this container is alive. Containers that
+	// fail liveness probes will be restarted.
+	// +optional
+	LivenessProbe *ContainerHealthProbe `json:"livenessProbe,omitempty"`
+
+	// A ReadinessProbe assesses whether this container is ready to serve
+	// requests. Containers that fail readiness probes will be withdrawn from
+	// service.
+	// +optional
+	ReadinessProbe *ContainerHealthProbe `json:"readiessProbe,omitempty"`
+
+	// TODO(negz): Ideally the key within this secret would be configurable, but
+	// the current OAM spec allows only a secret name.
+
+	// ImagePullSecret specifies the name of a Secret from which the
+	// credentials required to pull this container's image can be loaded.
+	// +optional
+	ImagePullSecret *string `json:"imagePullSecret,omitempty"`
+}
+
+// A ContainerisedWorkloadSpec defines the desired state of a
+// ContainerisedWorkload.
+type ContainerisedWorkloadSpec struct {
+	// OperatingSystem required by this workload.
+	// +kubebuilder:validation:Enum=linux;windows
+	// +optional
+	OperatingSystem *OperatingSystem `json:"osType,omitempty"`
+
+	// CPUArchitecture required by this workload.
+	// +kubebuilder:validation:Enum=i386;amd64;arm;arm64
+	// +optional
+	CPUArchitecture *CPUArchitecture `json:"arch,omitempty"`
+
+	// Containers of which this workload consists.
+	Containers []Container `json:"containers"`
+}
+
+// A ResourceReference refers to an resource managed by an OAM resource.
+type ResourceReference struct {
+	// APIVersion of the referenced resource.
+	APIVersion string `json:"apiVersion"`
+
+	// Kind of the referenced resource.
+	Kind string `json:"kind"`
+
+	// Name of the referenced resource.
+	Name string `json:"name"`
+
+	// UID of the referenced resource.
+	// +optional
+	UID *types.UID `json:"uid,omitempty"`
+}
+
+// A ContainerisedWorkloadStatus represents the observed state of a
+// ContainerisedWorkload.
+type ContainerisedWorkloadStatus struct {
+	runtimev1alpha1.ConditionedStatus `json:",inline"`
+
+	// Resources managed by this containerised workload.
+	Resources []ResourceReference `json:"resources,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// A ContainerisedWorkload is a workload that runs OCI containers.
+// +kubebuilder:subresource:status
+type ContainerisedWorkload struct {
```

Comment:
Nitpick: can we call this `ContainerizedWorkload` to match updates to spec in https://github.com/oam-dev/spec/pull/316? I think this may actually be necessary as it is a core type and must be implemented as compliant with spec. Alternatively we could make the spec call it `ContainerisedWorkload`. The internet did not seem favor one or the other 🤷‍♂

Reply:
Yeah, will do.